### PR TITLE
Encode attachment names in template-based emails

### LINKED ISSUE #59: Strange behavior regarding attachment in DefaultEmailService using templates
When you send a message using DefaultEmailService.send(email, template, modelObject, inlinePictures), attachments are included twice in the message:
- The first time through method "DefaultEmailService.toMimeMessage", using Spring MimeMessageHelper.
- The second time at line 120, this time not using Spring MimeMessageHelper. This code is probably wrong, as the comment line 121 relates to image parts, not attachments.

Oddly, this does not cause the attachments to be added twice in the final email, but because MimeMessageHelper is not called the second time, MimeUtility.encodeText is not applied on attachment filename, which results in attachment filename to be missing in my case.

Workaround: call MimeUtility.encodeText myself:
`DefaultEmailAttachment.builder().attachmentName(MimeUtility.encodeText(filename)))`

I guess some refactoring related to inlinePictures and attachments would help :)

### PR DESCRIPTION
**Pull request by fromi**
Resolve #59 by encoding attachment names

### CODE REVIEW

File: spring-boot-email-core/src/main/java/it/ozimov/springboot/mail/service/defaultimpl/DefaultEmailService.java [hunk 1]
```diff
@@ -123,7 +124,7 @@ public MimeMessage send(final @NonNull Email email,
                 DataSource source = new ByteArrayDataSource(emailAttachment.getAttachmentData(),
                         emailAttachment.getContentType().toString());
                 attachmentPart.setDataHandler(new DataHandler(source));
-                attachmentPart.setFileName(emailAttachment.getAttachmentName());
+                attachmentPart.setFileName(MimeUtility.encodeText(emailAttachment.getAttachmentName()));
```
Comment (robertotru) [inline spring-boot-email-core/src/main/java/it/ozimov/springboot/mail/service/defaultimpl/DefaultEmailService.java]: As far as the doc says, only headers must be MIME encoded http://docs.oracle.com/javaee/6/api/javax/mail/internet/MimeUtility.html#encodeText(java.lang.String)

Read also: https://stackoverflow.com/questions/28452599/javamail-attachment-filename-not-displaying-utf-8-characters-correctly

Anyway, if we want to go for this solution, then we need to set the same encoding as the email.
``Email`` class has ``getEncoding()``.

Maybe it should be ``MimeUtility.encodeText(emailAttachment.getAttachmentName(), email.getEncoding(), null)``;
But unit tests should confirm the former or the latter approach.

Comment (fromi) [inline spring-boot-email-core/src/main/java/it/ozimov/springboot/mail/service/defaultimpl/DefaultEmailService.java]: For my project, this is the difference between having a file name or not having a file name. Unfortunately I cannot explain why :(
If you look at Spring MimeMessageHelper line 996, you will see that they do encode attachments file name, using default system encoding. You are using this code for emails not based on templates.
You are probably right about using the email encoding, I guess MimeMessageHelper.addAttachment is not the best piece of code in Spring ;)

Comment (robertotru) [inline spring-boot-email-core/src/main/java/it/ozimov/springboot/mail/service/defaultimpl/DefaultEmailService.java]: Ok I will do some tests.

**Comment by robertotru** [comment]
Release 0.6.3 is out with your change.

**Pull request merged**